documentation/docs/05-load.md

```diff
@@ -93,85 +146,176 @@ An instance of [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL), co
 
 ```js
-/// file: src/routes/+page.js
-// @filename: ambient.d.ts
-declare module '$lib/api' {
-	interface Data{}
-	export const base: string;
-	export const client: {
-		get: (resource:string) => Promise<Data>
+/// file: src/routes/blog/+page.js
+import { browser } from '$app/environment';
+
+// this shared variable is safe to modify on the browser, as it only affects the current user
+let count = 0;
+
+/** @type {import('./$types').PageLoad} */
+export async function load() {
+	// Increase count each time the load function is run, but only in the browser
+	const returnedCount = count;
+	if (browser) {
+		count++;
 	}
+	return { count: returnedCount };
 }
+```
+
+Sometimes it's even benefitial to use both `load` functions. For example you could first return some data from your database, and use that to create an object that is nonserializable and therefore couldn't be returned from the server `load` function. To access the server data in the shared `load` function, use the `data` input:
+
+```js
+/// file: src/routes/blog/+page.server.js
+// @filename: ambient.d.ts
+declare const db: { getAllBlogPosts: () => Promise<any[]> }
 
 // @filename: index.js
 // ---cut---
-import * as api from '$lib/api';
+/** @type {import('./$types').PageServerLoad} */
+export async function load() {
+	return {
+		posts: await db.getAllBlogPosts()
+	};
+}
+```
 
-/** @type {import('./$types').PageLoad} */
-export async function load({ depends }) {
-	depends(
-		`${api.base}/foo`,
-		`${api.base}/bar`,
-		'my-stuff:foo'
-	);
+```js
+/// file: src/routes/blog/+page.js
+// @filename: ambient.d.ts
+declare const Post: any;
 
+// @filename: index.js
+// @errors: 2531 7006
+// ---cut---
+/** @type {import('./$types').PageLoad} */
+export async function load({ data }) {
 	return {
-		foo: api.client.get('/foo'),
-		bar: api.client.get('/bar')
+		posts: data.posts.map(post => new Post(post))
 	};
 }
 ```
 
 ```svelte
-/// file: src/routes/+page.svelte
+/// file: src/routes/blog/+page.svelte
 <script>
-	import { invalidate } from '$app/navigation';
-
 	/** @type {import('./$types').PageData} */
 	export let data;
-
-	const pageRefresh = async () => {
-		await invalidate('my-stuff:foo');
-	}
 </script>
 
-<p>{data.foo}<p>
-<p>{data.bar}</p>
-<button on:click={pageRefresh}>Refresh my stuff</button>
+{#each data.posts as post}
+	...
+{/each}
 ```
 
-#### fetch
+To summarize:
+- use server-only `load` in `+page.server.js`/`+layout.server.js` when
+	- you want to access your data base directly
+	- you access private environment variables
+	- the returned data is serializable
+- use shared `load` in `+page.js`/`+layout.js` when
+	- you want more control over the `load` function
+	- you want to avoid a network hop when the resource is on a different server
+	- you need to return non-serializable data
+	- you are building an SPA served from a static file server
+- use both `load` functions in case you have a combination of requirements
+
+### Making fetch requests
```

Comment:
@Conduitry in #7113 you mentioned you wanted some more details about the subtle difference of `fetch` and when to use it - this would be the section to add it to.